### PR TITLE
Update support docs to java11 and min version 64

### PR DESCRIPTION
**Pull request by jonathanmos**
## Description
Update documentation to reflect that we now use java11 and that the minimum RN version has changed to 0.64

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -66,9 +66,9 @@ Detox is built from the ground up to support React Native projects as well as pu
...
-| ≤0.66.x | ≥0.59, ≤0.66 - Visibility edge-case: see this [RN issue](https://github.com/facebook/react-native/issues/23870)\* |
+| iOS                                                                                            | Android                                                                                                                                                                                                                                        |
+|------------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| ≤0.66.x                                                                                        | ≥0.64, ≤0.66 - <ul> <li>Visibility edge-case: see this [RN issue](https://github.com/facebook/react-native/issues/23870) </li><li>Switch edge-case: see this [RN issue](https://github.com/facebook/react-native/issues/23870) </li></ul> |
```
Comment (asafkorem) [inline README.md:71]: you used the wrong url for the switch edge case link

**Review by asafkorem — CHANGES_REQUESTED** [review]

**Comment by jonathanmos** [comment]
retest this please

**Comment by jonathanmos** [comment]
retest this please

**Comment by jonathanmos** [comment]
retest this please

**Comment by jonathanmos** [comment]
retest this please

**Comment by jonathanmos** [comment]
retest this please

File: README.md [hunk 2]
```diff
@@ -66,9 +66,9 @@ Detox is built from the ground up to support React Native projects as well as pu
...
-| ≤0.66.x | ≥0.59, ≤0.66 - Visibility edge-case: see this [RN issue](https://github.com/facebook/react-native/issues/23870)\* |
+| iOS                                                                                            | Android                                                                                                                                                                                                                                        |
+|------------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| ≤0.66.x                                                                                        | ≥0.64, ≤0.66 - <ul> <li>Visibility edge-case: see this [RN issue](https://github.com/facebook/react-native/issues/23870) </li><li>Switch edge-case: see this [RN issue](https://github.com/facebook/react-native/issues/32594) </li></ul> |
```
Comment (d4vidi) [inline README.md:71]: I don't think the switch edge-case should be mentioned here, because we've managed to work our way around it.

File: docs/Introduction.AndroidDevEnv.md [hunk 3]
```diff
@@ -15,26 +15,26 @@ Note that running automated UI tests is _not the same_ as developing Android app
...
+What needs to be verified is that `java` is in-path and that the output contains something like this:
 
 ```sh
-java version "1.8.0_121"
+java version "11.0.11"
 ...
 ```
 
-Namely, that the version is `1.8.x_abc`.
+Namely, that the version is `11.0.11`.
```
Comment (d4vidi) [inline docs/Introduction.AndroidDevEnv.md:33]: Better use a version wildcard-ish notation instead of specified numbers.

File: docs/Introduction.AndroidDevEnv.md [hunk 4]
```diff
@@ -15,26 +15,26 @@ Note that running automated UI tests is _not the same_ as developing Android app
```
Comment (d4vidi) [inline docs/Introduction.AndroidDevEnv.md:15]: Since RN 66 may not be the golden standard out there just yet, we should, at least for a while, explain that Java11 is required in order to support RN66+Android12, and that otherwise, J8 should be used, etc.

**Review by d4vidi — CHANGES_REQUESTED** [review]
Please review

File: docs/Introduction.AndroidDevEnv.md [hunk 5]
```diff
@@ -15,7 +15,7 @@ Note that running automated UI tests is _not the same_ as developing Android app
 
 This is the most basic step in the process, as without a proper Java SDK installed, nothing Android-ish works -- at least not from command-line, which is mandatory for executing `Detox`.
 
-_The bottom line is that **Android needs Java 11 installed**._
+_The bottom line is that **Android needs Java installed**. If you want to run with React Native 66 and Android 12 then it needs to be at least Java 11, otherwise you should have at least Java 8._
```
Comment (d4vidi) [inline docs/Introduction.AndroidDevEnv.md:18]: At least java 8? I think it's exactly java 8 (namely, java 9 won't work... or will it?)

**Review by d4vidi — APPROVED** [review]

**Comment by d4vidi** [comment]
@jonathanmos please revisit and merge when green
